tests/test_bitinformation_pipeline.py

```diff
@@ -6,24 +6,35 @@
 import xbitinfo as xb
 
 
-def test_full():
+@pytest.mark.parametrize(
+    "ds,dim,axis",
+    [
+        (pytest.lazy_fixture("ugrid_demo"), None, -1),
+        (pytest.lazy_fixture("icon_grid_demo"), "ncells", None),
+        (pytest.lazy_fixture("air_temperature"), "lon", None),
+        (pytest.lazy_fixture("rasm"), "x", None),
+        (pytest.lazy_fixture("ROMS_example"), "eta_rho", None),
+        (pytest.lazy_fixture("era52mt"), "time", None),
+        (pytest.lazy_fixture("eraint_uvz"), "longitude", None),
+    ],
+)
+def test_full(ds, dim, axis):
     """Test xbitinfo end to end."""
-    label = "air_temperature"
-    ds = xr.tutorial.load_dataset(label)
     # xbitinfo
-    bitinfo = xb.get_bitinformation(ds, dim="lon")
+    bitinfo = xb.get_bitinformation(ds, dim=dim, axis=axis)
     keepbits = xb.get_keepbits(bitinfo)
     # ds_bitrounded = xb.jl_bitround(ds, keepbits)
     ds_bitrounded = xb.xr_bitround(ds, keepbits)  # identical
     # save
+    label = "file"
     ds.to_netcdf(f"{label}.nc")
     ds.to_compressed_netcdf(f"{label}_compressed.nc")
     ds_bitrounded.to_compressed_netcdf(f"{label}_bitrounded_compressed.nc")
     # check size reduction
     ori_size = os.path.getsize(f"{label}.nc")
     compressed_size = os.path.getsize(f"{label}_compressed.nc")
     bitrounded_compressed_size = os.path.getsize(f"{label}_bitrounded_compressed.nc")
-    assert compressed_size < ori_size
+    assert compressed_size <= ori_size * 1.1
```

Comment:
Alright, I checked, and the example dataset `ROMS_example` is already compressed with *zlib* and shuffled including the coordinates. Although `to_compressed_netcdf` does not set any compression for the coordinates, it uses the one of the input file. Can we maybe exclude this dataset from the test? I feel like it is a better test when `compressed_size` has to be smaller than `ori_size`, otherwise it won't fail if the compression fails.

Reply:
> compressed_size has to be smaller than ori_size

either we skip this dataset or we allow slightly larger only for this file because it is zlib and shuffled

Reply:
I'm fine with both solutions.